app/services/scene-collections/scene-collections.ts

```diff
@@ -0,0 +1,865 @@
+    if (!this.state.activeId) return;
+    await this.saveCurrentApplicationStateAs(this.state.activeId);
+    this.SET_MODIFIED(this.state.activeId, (new Date()).toISOString());
+  }
+
+  /**
+   * This is a safe method that will load the requested scene collection.
+   * It is responsible for cleaning up and saving any existing config,
+   * setting the app in the apropriate loading state, and updating the state
+   * and server.
+   * @param id The id of the colleciton to load
+   * @param shouldAttemptRecovery whether a new copy of the file should
+   * be downloaded from the server if loading fails.
+   */
+  async load(id: string, shouldAttemptRecovery = true): Promise<void> {
+    this.startLoadingOperation();
+    await this.deloadCurrentApplicationState();
+
+    try {
+      await this.loadCollectionIntoApplicationState(id);
+      await this.setActiveCollection(id);
+    } catch (e) {
+      if (shouldAttemptRecovery) {
+        await this.attemptRecovery(id);
+      } else {
+        console.warn(`Unsuccessful recovery of scene collection ${id} attempted`);
+        await this.create();
+      }
+    }
+
+    this.finishLoadingOperation();
+  }
+
+  /**
+   * Creates and switches to a new blank scene collection
+   * @param name the name of the collection
+   * @param setupFunction a function that can be used to set
+   * up some state.  This should really only be used by the OBS
+   * importer.
+   */
+  async create(name?: string, setupFunction?: () => void): Promise<void> {
+    this.startLoadingOperation();
+    await this.deloadCurrentApplicationState();
+
+    if (setupFunction && setupFunction()) {
+      // Do nothing
+    } else {
+      this.setupEmptyCollection();
+    }
+
+    name = name || this.suggestName(DEFAULT_COLLECTION_NAME);
+    const id: string = uuid();
+
+    await this.insertCollection(id, name);
+    await this.setActiveCollection(id);
+    this.finishLoadingOperation();
+  }
+
+  /**
+   * Deletes the current scene collection
+   */
+  async delete(): Promise<void> {
+    this.startLoadingOperation();
+    await this.deloadCurrentApplicationState();
+    await this.removeCollection(this.state.activeId);
+
+    if (this.collections.length > 0) {
+      this.load(this.collections[0].id);
+    } else {
+      this.create();
+    }
+  }
+
+  /**
+   * Renames the current scene collection
+   * @param name the name of the new scene collection
+   */
+  rename(name: string) {
+    this.RENAME_COLLECTION(this.state.activeId, name, (new Date()).toISOString());
+  }
+
+  /**
+   * Calls sync, but will never cause a rejected promise.
+   * Instead, it will log an error and continue.
+   */
+  async safeSync() {
+    try {
+      await this.sync();
+    } catch (e) {
+      console.error('Scene collection sync failed: ', e);
+    }
+  }
+
+  /**
+   * Duplicates the current scene collection
+   * @param name the name of the new scene collection
+   */
+  async duplicate(name: string) {
```

Comment:
it is not obvious that this method also switch the scene collection. Maybe rename it to something like `duplicateAndSetActive` ?